strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/AuthenticatorsRegistry.java

```diff
@@ -0,0 +1,139 @@
+package org.carlspring.strongbox.authentication.registry;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+import org.carlspring.strongbox.authentication.registry.support.AuthenticatorsRegistrySerializer;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@JsonSerialize(using = AuthenticatorsRegistrySerializer.class)
+public class AuthenticatorsRegistry
+        implements Supplier<List<Authenticator>>
+{
+
+    final transient ReentrantLock lock = new ReentrantLock();
+
+    private final List<Authenticator> authenticators;
+
+    private List<Authenticator> lastSnapshot;
+
+    @Inject
+    private Optional<ObjectMapper> objectMapper;
+
+    public AuthenticatorsRegistry()
+    {
+        this(Collections.emptyList());
+    }
+
+
+    public AuthenticatorsRegistry(List<Authenticator> authenticators)
+    {
+        this.authenticators = new ArrayList<>(authenticators);
+        updateSnapshot();
+    }
+
+    /**
+     * Reloads the registry by replacing all authenticators using
+     * given collection.
+     *
+     * @param authenticators new collection of authenticators
+     */
+    public void reload(Collection<Authenticator> authenticators)
+    {
+        Objects.requireNonNull(authenticators, () -> "Required non-null replacing authenticators.");
+        final Lock lock = this.lock;
```

Comment:
I don't need it, @carlspring. First time I wasn't using additional local variable, but FindBugs started screaming that 

>  Multi-threading - Method does not release lock on all exception paths 

and I saw this tip: https://dev.carlspring.org/sonar/coding_rules#rule_key=findbugs%3AUL_UNRELEASED_LOCK_EXCEPTION_PATH 
when there is an additional local variable ... I thought maybe this "hint" will shut the Findbugs up :P I'll get rid of it